#Problème 3




#Importer les outils

In [0]:
# vecteurs et algèbre linéraire
import numpy as np
# les outils pour visulaliser
import matplotlib.pyplot as plt
# le solveur
import scipy.integrate as solveur

#Définir les constantes du problème

In [0]:

CA0 = 1.5 #mol/l
CB0 = 2  #mol/l
CC0 = 0  #mol/l
CD0 = 0  #mol/l
CE0 = 0  #mol/l
CF0 = 0  #mol/l
Vdot = 10 #l/min


#Définir les équations à résoudre

In [0]:
def equations(variables,V,Vdot): 
  FA,FB,FC,FD,FE,FF=variables 
  
  CA=FA/Vdot #mol/l
  CB=FB/Vdot #mol/l
  CC=FC/Vdot #mol/l
  CD=FD/Vdot #mol/l
  CE=FE/Vdot #mol/l
  CF=FF/Vdot #mol/l
  
  r1= 0.25*CA*CB #mol/l*min
  r2= 0.1*CA*CD #mol/l*min
  r3= 5*(CB**2)*CC#mol/l*min
  
  RA= -r1-3*r3 #mol/l*min
  RB= -2*r1-r3 #mol/l*min
  RC= r1+r2-2*r3 #mol/l*min
  RD= r1-2*r2+r3 #mol/l*min
  RE = r2 #mol/l*min
  RF = r3 #mol/l*min
  
  dFAdV= RA
  dFBdV= RB
  dFCdV= RC
  dFDdV= RD
  dFEdV = RE
  dFFdV = RF
  
  return dFAdV,dFBdV,dFCdV,dFDdV,dFEdV,dFFdV 

#Définir les conditions initiales

In [0]:
conditions_initiales=[CA0*Vdot,CB0*Vdot,CC0*Vdot,CD0*Vdot,CE0*Vdot,CF0*Vdot] 

#Résoudre le problème à l'aide du solveur

In [0]:
V=np.linspace(0,50,51) #définir les bornes d'intégration
resultats=solveur.odeint(equations,conditions_initiales,V,(Vdot,)) # la première colonne de ''resultat'' est FA

#Recherche du taux de conversion de A

In [0]:
FA=resultats[:,0] #mol/min    j'extrais les valeurs de FA



In [37]:
CA=FA/Vdot #mol/l
X= (CA0-CA)/CA0 #[-]      calcul du taux de conversion
X[V>=50][0]*100 #% et est la réponse au problème (on cherche le taux de conversion de A lorsque le volume vaut 50 l)

96.43072873188254

**Le taux de conversion de A est de 96.43%**

#Recherche de la concentration de C

In [39]:
FC=resultats[:,2] #mol/min     ici j'extrais les valeurs de FC
CC=FC/Vdot #mol/l
CC[V>=50][0] #mol/l  et est la réponse au problème (on cherche la concentration de C lorsque le volume vaut 50 l)

0.0034915183106730948

**La concentration de C est de 0.00349 mol/l**